cftime/_cftime.pyx

```diff
@@ -364,8 +364,18 @@ def cast_to_int(num):
             raise OverflowError('time values outside range of 64 bit signed integers')
         if isinstance(num, np.ma.core.MaskedArray):
             int_num = np.ma.masked_array(np.rint(num), dtype=np.int64)
+            # use ceil instead of rint if 1 microsec less than a second
+            # or floor if 1 microsec greater than a second (issue #187)
+            int_num = np.ma.where(int_num%10000000 == 1, \
```

Comment:
These should be 10^6, not 10^7, correct?
```suggestion
            int_num = np.ma.where(int_num%1000000 == 1, \
```

Reply:
Right - good catch.  I've also added a check on units, so this is only done if units are not microsecs or millisecs (where this kind of rounding is probably not desirable).